### PR TITLE
Add GDS-SSO OAuth credentials for govuk-chat

### PR DESCRIPTION
**Pull request by AlexAvlonitis**
trello card: https://trello.com/c/k8KTyMCx/1413-add-gds-sso-to-integration-environment

### CODE REVIEW

File: charts/app-config/values-integration.yaml [hunk 1]
```diff
@@ -1260,6 +1260,16 @@ govukApplications:
             secretKeyRef:
               name: govuk-chat-rabbitmq
               key: RABBITMQ_URL
+        - name: GDS_SSO_OAUTH_ID
+          valueFrom:
+            secretKeyRef:
+              name: govuk-chat-oauth
+              key: GDS_SSO_OAUTH_ID
```
Comment (kevindew) [inline charts/app-config/values-integration.yaml:1267]:
```suggestion
              name: signon-app-govuk-chat
              key: oauth_id
```

File: charts/app-config/values-integration.yaml [hunk 2]
```diff
@@ -1260,6 +1260,16 @@ govukApplications:
...
+        - name: GDS_SSO_OAUTH_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: govuk-chat-oauth
+              key: GDS_SSO_OAUTH_SECRET
```
Comment (kevindew) [inline charts/app-config/values-integration.yaml:1272]:
```suggestion
              name: signon-app-govuk-chat
              key: oauth_secret
```

File: charts/external-secrets/templates/govuk-chat/oauth.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+apiVersion: external-secrets.io/v1beta1
```
Comment (kevindew) [inline charts/external-secrets/templates/govuk-chat/oauth.yaml:1]: It doesn't look like other apps need this for signon credentials. I'm not sure how they get them but I have vague memories of there being a system that auto rotates them.

**Review by kevindew — COMMENTED** [review]
It looks like there's a bit of a pre-existing convention about signon credentials are managed. I've made some suggestions but I don't quite understand how it works so I'm not sure I've given you the right suggestions.

**Comment by kevindew** [comment]
Asked in: https://gds.slack.com/archives/C013F737737/p1714645006120649

File: charts/app-config/values-integration.yaml (hunk 1 above)
Comment (AlexAvlonitis) [reply charts/app-config/values-integration.yaml:1267]: The name shouldn't change as i think it's a convention to use `app-name + secret` https://docs.publishing.service.gov.uk/kubernetes/manage-app/manage-secrets/#defining-an-externalsecret-in-kubernetes 

but I can change the key name as you suggested 👍

Comment (kevindew) [reply charts/app-config/values-integration.yaml:1267]: It's just that all the other apps have ones prefixed with signon-app as per: https://github.com/alphagov/govuk-helm-charts/blob/2faad1e20782fa4a567a395dbd3e0ad898b85375/charts/app-config/values-integration.yaml#L122-L131

File: charts/external-secrets/templates/govuk-chat/oauth.yaml (hunk 3 above)
Comment (AlexAvlonitis) [reply charts/external-secrets/templates/govuk-chat/oauth.yaml:1]: that's what i noticed too, I only noticed account api using it. Nice thanks for notifying the team

File: charts/app-config/values-integration.yaml (hunk 1 above)
Comment (AlexAvlonitis) [reply charts/app-config/values-integration.yaml:1267]: hmm good point, they are getting them from somewhere else

**Comment by sengi** [comment]
Yeah just follow the naming of the existing ones and you should be fine.

**Comment by sengi** [comment]
You'll need to add the app to the list here: https://github.com/alphagov/govuk-helm-charts/blob/main/charts/app-config/templates/signon-secrets-sync-configmap.yaml

There's probably nothing stopping us dropping that file now, though. I think we only introduced it because of the number of defunct applications in Signon that hadn't been cleaned up at the time. It's much tidier now so just importing everything is probably a good option.

**Comment by kevindew** [comment]
As per the Slack chat I'm going to go and create the application in Signon in each environment before we proceed with merging

**Comment by kevindew** [comment]
> > As per the Slack chat I'm going to go and create the application in Signon in each environment before we proceed with merging
> 
> Added one in integration already, just double check about the naming conventions etc.. so we could use same names in all the ENVs

Ok cool - I'll create the other ones and adjust integration if I have to for any consistency reasons

File: charts/app-config/templates/signon-secrets-sync-configmap.yaml [hunk 4]
```diff
@@ -7,6 +7,7 @@ data:
     [
       "Account API",
       "Asset Manager",
+      "Chat",
```
Comment (AlexAvlonitis) [inline charts/app-config/templates/signon-secrets-sync-configmap.yaml:10]: Added this one for now, if we choose a different name I will change it again

Comment (kevindew) [reply charts/app-config/templates/signon-secrets-sync-configmap.yaml:10]: We won't need this when https://github.com/alphagov/govuk-helm-charts/pull/2006 is merged (which I imagine will be in the next hour).

Thanks for adding it though.

**Comment by kevindew** [comment]
Looks like it's merged now 🎉 

I've created the signon entry in prod, it'll sync to staging tonight. We won't wait for that before merging though (as far as I can tell, from Chris' changes signon won't error without Chat).

I imagine we will need to run a rake task in signon to get these secrets populated. I'm not sure on the automation schedule @sengi do you know?

**Review by kevindew — APPROVED** [review]
Great, looks good to me 🤞 that it all works.

**Comment by AlexAvlonitis** [comment]
> These values won't have `govuk-` as a prefix because they're based on the signon name, which is just chat.
> 
> I think with that changed this will be good to be merged.

Done, thanks for the reviews 👍

**Pull request merged**